Sources/Controllers/MessagesViewController.swift

```diff
@@ -251,49 +251,77 @@ extension MessagesViewController: UICollectionViewDataSource {
         return messagesLayoutDelegate.footerViewSize(for: message, at: indexPath, in: messagesCollectionView)
     }
 
+    open override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+
+        guard let lastVisibleIndexPath = messagesCollectionView.indexPathsForVisibleItems.sorted().last else { return }
+
+        coordinator.animate(alongsideTransition: { (context:UIViewControllerTransitionCoordinatorContext) in
+            self.messagesCollectionView.scrollToItem(at: lastVisibleIndexPath, at: .top, animated: false)
+        }, completion: { _ in
+            // For some reason, after animated scrolling we can scroll a bit more
+            self.messagesCollectionView.scrollToItem(at: lastVisibleIndexPath, at: .top, animated: true)
+        })
+    }
 }
 
 // MARK: - Keyboard Handling
 
 fileprivate extension MessagesViewController {
 
     func addKeyboardObservers() {
-        NotificationCenter.default.addObserver(self, selector: #selector(handleKeyboardDidChangeState), name: .UIKeyboardWillChangeFrame, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(handleKeyboardWillChangeState), name: .UIKeyboardWillChangeFrame, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(handleKeyboardDidChangeState), name: .UIKeyboardDidChangeFrame, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(handleTextViewDidBeginEditing), name: .UITextViewTextDidBeginEditing, object: messageInputBar.inputTextView)
     }
 
     func removeKeyboardObservers() {
         NotificationCenter.default.removeObserver(self, name: .UIKeyboardWillChangeFrame, object: nil)
+        NotificationCenter.default.removeObserver(self, name: .UIKeyboardDidChangeFrame, object: nil)
         NotificationCenter.default.removeObserver(self, name: .UITextViewTextDidBeginEditing, object: messageInputBar.inputTextView)
     }
 
     @objc
-    func handleTextViewDidBeginEditing(_ notification: Notification) {
+    private func handleTextViewDidBeginEditing(_ notification: Notification) {
```

Comment:
This doesn't need `private`

Reply:
`private` modifier makes those methods inaccessible from outside of the extension (if I'm not mistaken). With `fileprivate extension` - other extensions of this class can invoke the methods which could be undesirable.

But, I don't think it's that important to keep those methods as close as possible, `fileprivate extension` also works :)
I can remove it.

Reply:
Still not updated.